packages/konsole-app/src/hooks/useResources.ts

```diff
@@ -0,0 +1,20 @@
+//
+// Copyright 2021 DXOS.org
+//
+
+import { useEffect, useState } from 'react';
+
+import { IQuery, Resource } from '@dxos/registry-api';
+
+import { useRegistryClient } from './useRegistry';
+
+export function useResources (query?: IQuery): Resource[] {
+  const [resources, setResources] = useState<Resource[]>([]);
+  const registryClient = useRegistryClient();
+
+  useEffect(function () {
+    void registryClient.registry.getResources(query).then(setResources);
```

Comment:
We try to avoid using `.then` in favour of `async/await` whenever possible. I understand though that it will make this code uglier. I call this "HAJ" rule